### PR TITLE
feat: add example project highlighting basic functionality

### PR DESCRIPTION
**Pull request by alanorwick**

### CODE REVIEW

File: examples/example-project/README.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+```bash
+npm install
+# or
+yarn install
+```
+
+## Usage
+
+```bash
+npm start
```
Comment (alejoacosta74) [inline examples/example-project/README.md:30]: Cam you add the `start` script to `package.json?

**Review by alejoacosta74 — CHANGES_REQUESTED** [review]

File: examples/example-project/package.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "example-project",
+  "version": "1.0.0",
+  "main": "index.js",
+  "scripts": {
+    "start": "ts-node src/index.ts",
```
Comment (alejoacosta74) [inline examples/example-project/package.json:6]: `npm run start` is not working.
It looks like you need to add:
```json
  "type": "module",
```

and modify the `start` script to:
```json
    "start": "node --loader ts-node/esm src/index.ts",
```

**Review by alejoacosta74 — CHANGES_REQUESTED** [review]

File: examples/example-project/package.json (hunk 2 above)
Comment (alanorwick) [reply examples/example-project/package.json:6]: It works for me, what do you see?

Comment (alanorwick) [reply examples/example-project/package.json]: switched to tsx instead of ts-node

**Pull request closed without merging**